lightning/src/ln/peer_handler.rs

```diff
@@ -459,6 +459,16 @@ impl<Descriptor: SocketDescriptor, CM: Deref, L: Deref> PeerManager<Descriptor,
 		}
 	}
 
+	/// Append a message to a peer's pending outbound/write buffer, and update the map of peers needing sends accordingly
+	fn enqueue_message<M: Encode + Writeable>(&self, peers_needing_send: &mut HashSet<Descriptor>, peer: &mut Peer, descriptor: Descriptor, message: &M) {
+		let mut encoded_message: Vec<u8> = vec![];
+		wire::write(message, &mut encoded_message).unwrap(); // crash if the write failed
```

Comment:
I'd be surprised if this weren't a good bit slower in some cases than VecWriter. VecWriter implements the size_hint by allocating capacity in the Vec, which we use extensively in the message serialization path.

Reply:
Our default implementation of `Writer` delegates to `Write`. And then the `Write` implementation for `Vec<u8>` calls `extend_from_slice`, which uses `Iterator`'s `size_hint` to allocate enough space.

The implementation of `Writer` for `VecWriter` seems to be reinventing this. Or worse it's attempting to reserve enough capacity twice: once on its own in `size_hint` and then again when calling `extend_from_slice` on the underlying vector (as per above).

Reply:
yeah, we might wanna abandon VecWriter in this file altogether

Reply:
I suppose with `VecWriter` you only need to allocate once, though, if there is a size hint for the entire message vs needing to allocate for each of the message's component parts.

Reply:
Right, `VecWriter` does it at a higher level - we tell it how long the entire message is going to be vs as we write individual fields. This should avoid quite a few allocations in our serialization path, which tends to be a common performance issue.

Reply:
Per your original comment, I had switched it to VecWriter, so we can just keep it.